browsers/duckduckgo.safariextension/js/abp-preprocessed.es6.js

```diff
@@ -0,0 +1,135 @@
+    }
+}
+
+// these are defined in trackers.js
+easylists = lists.easylists
+whitelists = lists.whitelists
+
+/*
+ * Get the list data and use abp to parse.
+ * The parsed list data will be added to 
+ * the easyLists object.
+ */
+function updateLists () {
+    const atb = settings.getSetting('atb')
+    const set_atb = settings.getSetting('set_atb')
+    const versionParam = getVersionParam()
+
+    for (let listType in lists) {
+        for (let name in lists[listType]) {
+
+            const constantsName = lists[listType][name].constantsName
+            
+            let url = constants[constantsName]
+            if (!url) return 
+                
+            let etag = settings.getSetting(constantsName + '-etag') || ''
+
+            // only add url params to contentblocking.js duckduckgo urls
+            if(url.match(/^https?:\/\/(.+)?duckduckgo.com\/contentblocking\.js/)) {
+                if (atb) url += '&atb=' + atb
+                if (set_atb) url += '&set_atb=' + set_atb
+                if (versionParam) url += versionParam
+            }
+
+            console.log(`Checking for list update: ${name} ${url} ${etag}`)
+
+            // if we don't have parsed list data skip the etag to make sure we
+            // get a fresh copy of the list to process
+            if (Object.keys(lists[listType][name].parsed).length === 0) etag = ''
+                
+            load.loadExtensionFile({url: url, source: 'external', etag: etag}, (listData, response) => {
+                const newEtag = response.getResponseHeader('etag') || ''
+                console.log('Updating list: ', name)
+
+                // sync new etag to storage
+                settings.updateSetting(constantsName + '-etag', newEtag)
+                
+                abp.parse(listData, lists[listType][name].parsed)
+                lists[listType][name].isLoaded = true
+            })
+        }
+    }
+
+    let trackersWhitelistTemporaryEtag = settings.getSetting('trackersWhitelistTemporary-etag') || ''
+    // reset etag to get a new list copy if we don't have brokenSiteList data
+    if (!trackersWhitelistTemporary || !trackersWhitelistTemporaryEtag) trackersWhitelistTemporaryEtag = ''
+
+    // load broken site list
+    // source: https://github.com/duckduckgo/content-blocking-whitelist/blob/master/trackers-whitelist-temporary.txt
+    load.loadExtensionFile({url: constants.trackersWhitelistTemporary, etag: trackersWhitelistTemporaryEtag, source: 'external'}, (listData, response) => {
+        const newTrackersWhitelistTemporaryEtag = response.getResponseHeader('etag') || ''
+        settings.updateSetting('trackersWhitelistTemporary-etag', newTrackersWhitelistTemporaryEtag);
+
+        // defined in site.js
+        trackersWhitelistTemporary = listData.trim().split('\n')
+    })
+}
+
+// Make sure the list updater runs on start up
+settings.ready().then(() => updateLists())
+
+// 3 hour update interval
+setInterval(updateLists, 3*60*60*1000)
```

Comment:
we modified this to 30 minutes:
https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/24/files

Reply:
actually, I think we can merge this file into the shared one pretty easily, I'm going to open a PR against this branch for that.

Reply:
Great! Thanks

Reply:
https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/40